src/test/java/com/gfl/client/model/ProxyCredentialsTest.java

```diff
@@ -0,0 +1,176 @@
+package com.gfl.client.model;
+
+import jakarta.validation.ConstraintViolation;
+import jakarta.validation.Validation;
+import jakarta.validation.Validator;
+import jakarta.validation.ValidatorFactory;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import java.util.Set;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+
+class ProxyCredentialsTest {
+    private ProxyCredentials proxyCredentials;
+    private final ValidatorFactory validatorFactory = Validation.buildDefaultValidatorFactory();
+    private final Validator validator = validatorFactory.getValidator();
+
+    @BeforeEach
+    void setUp(){
+        proxyCredentials = new ProxyCredentials("username1", "password1");
+    }
+
+    @Test
+    @DisplayName("Test empty constructor with null password and username")
+    void testEmptyConstructor(){
+        ProxyCredentials emptyProxyCredentials = new ProxyCredentials();
+
+        assertNull(emptyProxyCredentials.getUsername());
+        assertNull(emptyProxyCredentials.getPassword());
+    }
+
+    @Test
+    @DisplayName("Test getting password from ProxyCredentials object")
+    void testProxyCredentialsGetPassword(){
+        assertEquals("password1", proxyCredentials.getPassword());
+    }
+
+    @Test
+    @DisplayName("Test getting username from ProxyCredentials object")
+    void testProxyCredentialsGetUsername(){
+        assertEquals("username1", proxyCredentials.getUsername());
+    }
+
+    @Test
+    @DisplayName("Test setting username from ProxyCredentials object")
+    void testProxyCredentialsSetUsername(){
+        assertEquals("username1", proxyCredentials.getUsername());
+        proxyCredentials.setUsername("username");
+        assertEquals("username", proxyCredentials.getUsername());
+    }
+
+    @Test
+    @DisplayName("Test setting password from ProxyCredentials object")
+    void testProxyCredentialsSePassword(){
+        assertEquals("password1", proxyCredentials.getPassword());
+        proxyCredentials.setPassword("password");
+        assertEquals("password", proxyCredentials.getPassword());
+    }
+
+    @Test
+    @DisplayName("Test equality between ProxyCredentials objects")
+    void testEquals() {
+        ProxyCredentials SameProxyCredentials = new ProxyCredentials("username1", "password1");
+        assertEquals(proxyCredentials, SameProxyCredentials);
+    }
+
+    @Test
+    @DisplayName("Test not equality between ProxyCredentials objects")
+    void testEqualsFailed() {
+        ProxyCredentials SameProxyCredentials = new ProxyCredentials("username", "password");
+        assertNotEquals(proxyCredentials, SameProxyCredentials);
+    }
+
+    @Test
+    @DisplayName("Test hashCode method for ProxyCredentials object")
```

Comment:
If you want to ensure that the hashCode method adheres to its contract, 
you could write a more comprehensive test by checking:

Consistency: Multiple invocations of hashCode on the same object should consistently return the same hash code.
Equality: If two objects are equal (according to the equals method), their hash codes should be the same.

```
@Test
@DisplayName("Test hashCode method for ProxyCredentials object")
void testHashCode() {
    // Consistency: Multiple invocations should return the same hash code
    int firstHashCode = proxyCredentials.hashCode();
    int secondHashCode = proxyCredentials.hashCode();
    assertEquals(firstHashCode, secondHashCode);

    // Equality: If two objects are equal, their hash codes should be the same
    ProxyCredentials equalCredentials = new ProxyCredentials("username", "password");
    assertEquals(proxyCredentials, equalCredentials);
    assertEquals(proxyCredentials.hashCode(), equalCredentials.hashCode());
}
```